src/main/kotlin/parts/wisdom/arcraider/visualization/SwingApp.kt

```diff
@@ -1,22 +1,66 @@
 package parts.wisdom.arcraider.visualization
 
+import parts.wisdom.arcraider.Task
 import parts.wisdom.arcraider.loadTaskFromFile
+import java.awt.Dimension
 import java.awt.EventQueue
+import java.awt.FlowLayout
 import java.io.File
+import javax.swing.*
 
 private const val WINDOW_HEIGHT_PIXEL = 600
 private const val WINDOW_WIDTH_PIXEL = 800
 
+private var currentGridPanel : GridPane? = null
+
 private fun createAndShowGUI() {
     val frame = ARCWindow(WINDOW_WIDTH_PIXEL, WINDOW_HEIGHT_PIXEL, "ARCRaider")
     val task = loadTaskFromFile(
             File(object {}.javaClass.classLoader.getResource("extend_right.json").toURI()))
 
-    val grid = Grid(task.test[0].input)
-    frame.add(GridPane(grid))
+    val content = JPanel()
+    content.layout = BoxLayout(content, BoxLayout.Y_AXIS)
+
+    renderGrid(task)
+    val buttonPanel = renderButtonPanel(content)
+
+    content.add(buttonPanel)
+    content.add(currentGridPanel)
+    frame.add(content)
     frame.isVisible = true
 }
 
+private fun resetGrid(content: JPanel, task : Task) {
```

Comment:
Why do we create a new grid every time? Can't we just update the existing Panel?

Reply:
fixed